diesel/src/query_dsl/join_dsl.rs

```diff
@@ -88,7 +88,7 @@ where
 /// [`allow_tables_to_appear_in_same_query!`]: ../macro.allow_tables_to_appear_in_same_query.html
 pub trait JoinDsl: Sized {
     /// Join two tables using a SQL `INNER JOIN`. The `ON` clause is defined
-    /// via the [associations API](../associations/index.html).
+    /// via the [JoinDsl](./trait.JoinOnDsl.html#method.on).
```

Comment:
We should mention here that a default/implicit on clause is used if `JoinOnDsl::on` is not called.